src/crimson/tools/store_nbd/store-nbd.cc

```diff
@@ -321,7 +321,7 @@ class nbd_oldstyle_negotiation_t {
   uint64_t magic2 = seastar::cpu_to_be(0x00420281861253);  // "IHAVEOPT"
   uint64_t size = 0;
   uint32_t flags = seastar::cpu_to_be(0);
-  [[maybe_unused]] char reserved[124] = {0};
+  char reserved[124] = {0};
```

Comment:
i added this to silence warning from GCC. see 9f70239ef007392a9ce4c421fe944164fdad6ca0. how about set an option just for this file, something like
```cmake
  set_source_files_properties(
    store-nbd.cc
    APPEND APPEND_STRING
    COMPILE_FLAGS -Wno-unused-variable)
```

Reply:
Ah, [there is a discrepancy between the compilers](https://stackoverflow.com/a/65633590). `gcc version 10.2.1 20201112 (Red Hat 10.2.1-8)` was complaining about the `[[maybe_unused]]` and the removal hasn't caused any different warning to pop up in the file.

Other possible solutions:
* apply `-Wno-unused-private-field` globally as mentioned in SO response,
* make the `reserved` field non-`private` (`protected`?).

Reply:
in that case, i am fine either way. just wanted to save the effort reading the warning message under most circumstances.

Reply:
when you remove [[maybe_unused]], where you add COMPILE_FLAGS " -Wno-unused-variable" ?  it is not in the PR.

Reply:
when remove [[maybe_unused]], does it warn some variable defined but not used?  where add COMPILE_FLAGS " -Wno-unused-variable") ?  it it not in this PR.

Reply:
> when remove [[maybe_unused]], does it warn some variable defined but not used? 

@liu-chunmei guess i was using GCC-9. while it seems both GCC-9 and GCC-10 are fine with this. see https://godbolt.org/z/xnE88s4KM


> where add COMPILE_FLAGS " -Wno-unused-variable") ? it it not in this PR.

in CMakeLists.txt.